Test suite

ErlingWis opened the pull request:
Vær så snill å sammenligne med [de matematiske funksjonene](https://link.springer.com/content/pdf/10.1007%2Fs00500-019-03794-x.pdf)

ErlingWis committed:
Test Problem 1

ErlingWis committed:
removed env file

ErlingWis committed:
removed .env file

ErlingWis committed:
test function 2

ErlingWis committed:
LIR-CMOP1-14 v0.1

ErlingWis committed:
.gitignored bro

ErlingWis committed:
gitignore fix 2

mariusaarsnes commented on testSuite/structs.go:
```diff
@@ -0,0 +1,6 @@
+
+type Fitness struct {
+	Objectives, SoftConstraints map[string]float64
+	HardConstraints             map[string]bool
```
Kan constraints få andre navn enn soft of hard? Jeg (Erling) foreslår å bruke likhet- og ulikhet-beskrankning ettersom det gjenspeiler beskrankningene bedre

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+
+	return fitness
+}
+
```
ok

mariusaarsnes commented on testSuite/objectiveFunctions.go:
```diff
@@ -0,0 +1,51 @@
+}
+
+func objective3(x []float64) float64 {
+	return math.Sqrt(x[0]) + inner2(x)
```
```suggestion
	return 1 - math.Sqrt(x[0]) + inner2(x)
```

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C1"] = constraint1(x, inner1)
+	fitness.HardConstraints["C2"] = constraint1(x, inner2)
+
+	return fitness
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C3"] = constraint2(x)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C3"] = constraint2(x)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/constraints.go:
```diff
@@ -0,0 +1,29 @@
+
+func constraint3(x, p, q, a, b []float64, k int, f1, f2 float64) bool {
+	d := -0.25 * math.Pi
+	z := math.Pow((f1-p[k])*math.Cos(d)-(f2-q[k])*math.Sin(d), 2) / math.Pow(a[k], 2)
```
Kan flytte opphevingen ut av brøken

mariusaarsnes commented on testSuite/constraints.go:
```diff
@@ -0,0 +1,29 @@
+	return math.Sin(20*math.Pi*x[0])-0.5 >= 0
+}
+
+func constraint3(x, p, q, a, b []float64, k int, f1, f2 float64) bool {
```
trenger kanskje ikke sende inn x siden vi allerede bruker den utregnede verdien for f1 og f2 (dette burde nevnes i omdel/experiment kapittelet)

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint3(x, p, q, a, b, 1, fitness.Objectives["F1"], fitness.Objectives["F2"])
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint3(x, p, q, a, b, 1, fitness.Objectives["F1"], fitness.Objectives["F2"])
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C3"] = constraint3(x, p, q, a, b, 2, fitness.Objectives["F1"], fitness.Objectives["F2"])
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C3"] = constraint3(x, p, q, a, b, 2, fitness.Objectives["F1"], fitness.Objectives["F2"])
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint4(x, fitness.Objectives["F1"], fitness.Objectives["F2"], 2)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint4(x, fitness.Objectives["F1"], fitness.Objectives["F2"], 1)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint4(x, fitness.Objectives["F1"], fitness.Objectives["F2"], 2.1)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	fitness.HardConstraints["C2"] = constraint4(x, fitness.Objectives["F1"], fitness.Objectives["F2"], 2.5)
+
+	return fitness
+}
```
ok

mariusaarsnes commented on testSuite/objectiveFunctions.go:
```diff
@@ -0,0 +1,51 @@
+}
+
+func objective12(x []float64) float64 {
+	return (1.7057 + inner5(x)) * math.Cos(0.5*math.Pi*x[0])
```
sinus istedenfor cosinus

mariusaarsnes commented on testSuite/problems.go:
```diff
@@ -0,0 +1,246 @@
+	return fitness
+}
+
+func CMOP13(x []float64) Fitness {
+	fitness := Fitness{}
+
+	fitness.SoftConstraints = make(map[string]float64)
+	fitness.Objectives = make(map[string]float64)
+
+	fitness.Objectives["F1"] = objective10(x)
+	fitness.Objectives["F2"] = objective11(x)
+	fitness.Objectives["F3"] = objective12(x)
+
+	g := math.Pow(fitness.Objectives["F1"], 2) + math.Pow(fitness.Objectives["F2"], 2) + math.Pow(fitness.Objectives["F3"], 2)
+
+	fitness.SoftConstraints["C1"] = constraint5(x, g, 9, 4)
+	fitness.SoftConstraints["C2"] = constraint5(x, g, 3.61, 3.24)
+
+	return fitness
+}
+
+func CMOP14(x []float64) Fitness {
+	fitness := Fitness{}
+
+	fitness.SoftConstraints = make(map[string]float64)
+	fitness.Objectives = make(map[string]float64)
+
+	fitness.Objectives["F1"] = objective10(x)
+	fitness.Objectives["F2"] = objective11(x)
+	fitness.Objectives["F3"] = objective12(x)
+
+	g := math.Pow(fitness.Objectives["F1"], 2) + math.Pow(fitness.Objectives["F2"], 2) + math.Pow(fitness.Objectives["F3"], 2)
+
+	fitness.SoftConstraints["C1"] = constraint5(x, g, 9, 4)
+	fitness.SoftConstraints["C2"] = constraint5(x, g, 3.61, 3.24)
+	fitness.SoftConstraints["C3"] = constraint5(x, g, 3.0625, 2.56)
+
+	return fitness
+}
```
ok

mariusaarsnes reviewed (changes requested):
Et par småting som må rettes opp. Ser for det meste riktig ut

ErlingWis committed:
Bugfixes / enhancements

Fixed objective3() to correct formula
Enhanced constraint3() with outer polynomial
Fixed objective12() to correct formula

ErlingWis committed:
Refactor

Constraints renamed to
SoftConstraints -> Constraints that return a float value describing the constraint violation
InequalityConstraints -> Constraints that return a boolean describing if the constraint fulfils the requirements.

mariusaarsnes reviewed (approved):
ok jobba
